sklearn/utils/validation.py

```diff
@@ -374,6 +378,9 @@ def check_array(array, accept_sparse=None, dtype="numeric", order=None,
         if force_all_finite:
             _assert_all_finite(array)
 
+    if array.ndim == 1:
```

Comment:
I am a little confused myself now. Here is what I understand

* Warn against 1D arrays now and raise an error in later versions.
* If `ensure_2d` is set, nothing to be done now. In later versions `ensure_2d` parameter is deprecated ?